R/tspa_plot_revised.R

```diff
@@ -0,0 +1,123 @@
+tspa_plot <- function(tspa_fit,
+                      ask = FALSE,
+                      title = NULL,
+                      label_x = NULL,
+                      label_y = NULL,
+                      abbreviation = TRUE,
+                      fscore_type = NULL,
+                      ...) {
+
+  fit_pars <- lavaan::parameterestimates(tspa_fit) # parameter estimates
+  regression_fit <- fit_pars[which(fit_pars$op == "~"),]
+  latent_dv <- unique(c(unname(t(regression_fit["lhs"])))) # Extract DV
+  latent_iv <- unique(c(unname(t(regression_fit["rhs"])))) # Extract IV
+  fscores_func <- lavaan::lavInspect(tspa_fit, what = "data") # factor scores from `get_fs`
+  fscores_est <- lavaan::lavPredict(tspa_fit) # factor scores from estimation using `lavPredict`
+
+    # Type of scatterplot
+    if (is.null(fscore_type)) {
+      fscores <- fscores_est
+    } else {
+      fscores <- fscore_func
```

Comment:
The variable names do not match. And if you're using conditional statements, put the computation inside the if-else, so that you don't need to do unnecessary computations. E.g.,

`if (...) {
    fscores <- lavaan::lavInspect(...)
} else {
    fscores <- ...
}

Reply:
```
if (fscores_type == "original") {
      fscores <- lavaan::lavInspect(tspa_fit, what = "data") # factor scores from `get_fs`
    } else {
      fscores <- lavaan::lavPredict(tspa_fit) # factor scores from estimation using `lavPredict`
    }
```

Regarding "the variable names do not match", do you mean "fscores" and "fscore_type"?  Do you think adding a "s" solve this problem? Please let me know if I misunderstand anything. Thanks!